hotsos/core/plugins/openstack/common.py

```diff
@@ -214,6 +214,19 @@ def apache2_certificates_expiring(self):
 
         return apache2_certificates_expiring
 
+    @cached_property
+    def apache2_allow_encoded_slashes_on(self):
+        """ Returns True if AllowEncodedSlashes On is found. """
+        if not self.ssl_enabled:
+            return False
+
+        with open(self.apache2_ssl_config_file) as fd:
+            for line in fd:
+                ret = re.search(r'AllowEncodedSlashes On /(\S+)', line)
```

Comment:
The regex is wrong - that's why tests are failing. 

Since you don't look at anything after `On`, rest needs to be dropped. It's also better to account for multiple spaces between `AllowEncodedSlashes` and `On`. So it would be:
```
                ret = re.search(r'AllowEncodedSlashes\s+On', line)
```

Should it work for lowercase `on` too? In that case, you can use:
```
                ret = re.search(r'AllowEncodedSlashes\s+[Oo]n', line)
```

Likewise for `AllowEncodedSlashes`:
```
                ret = re.search(r'[Aa]llow[Ee]ncoded[Ss]lashes\s+[Oo]n', line)
```

Reply:
I did test these, although the build goes through and the tests pass, the resulting binary is not detecting the target LP when executed against my reference SOS report. Need to look deeper.

Reply:
@alanbach I can look take a look. Which sosreport it doesn't work for? Btw, you can run hotsos locally with your changes without building a binary.

Reply:
@pponnuvel I'm building the binary to make sure all the steps are clean and passed. 

I'm not quite sure if this regex actually works. I took a SOS report from one of my RGW units affected with this LP and running the built sosreport against it does not seem to detect and report the issue.

Reply:
I internally shared the SOS report

Reply:
@alanbach 

I ran it and it does detect:
```
storage:
  ceph:
    release:
      name: quincy
      days-to-eol: 2993
    services:
      systemd:
        enabled:
          - ceph-radosgw
        masked:
          - radosgw
      ps: 
        - radosgw (1) 
    dpkg:
      - ceph-common 17.2.6-0ubuntu0.22.04.3
      - python3-ceph-argparse 17.2.6-0ubuntu0.22.04.3
      - python3-ceph-common 17.2.6-0ubuntu0.22.04.3
      - python3-cephfs 17.2.6-0ubuntu0.22.04.3
      - python3-rados 17.2.6-0ubuntu0.22.04.3
      - python3-rbd 17.2.6-0ubuntu0.22.04.3
      - radosgw 17.2.6-0ubuntu0.22.04.3
  bugs-detected:
    https://bugs.launchpad.net/bugs/1974138: In multisite S3 RGW environments, objects
      containing prefixes (/), do not sync from primary to secondary sites if AllowEncodedSlashes
      Apache directive is missing or set to OFF.
````